packages/analytics-js-common/src/types/ErrorHandler.ts

```diff
@@ -1,19 +1,41 @@
 import type { IPluginEngine } from './PluginEngine';
 import type { ILogger } from './Logger';
 import type { IExternalSrcLoader } from '../services/ExternalSrcLoader/types';
+import type { BufferQueue } from '../services/BufferQueue/BufferQueue';
 
-export type SDKError = unknown;
+export type SDKError = unknown | Error | ErrorEvent | Event | PromiseRejectionEvent;
 
 export interface IErrorHandler {
   logger?: ILogger;
   pluginEngine?: IPluginEngine;
+  errorBuffer: BufferQueue<PreLoadErrorData>;
   init(externalSrcLoader: IExternalSrcLoader): void;
   onError(
     error: SDKError,
     context?: string,
     customMessage?: string,
     shouldAlwaysThrow?: boolean,
+    errorType?: string,
   ): void;
   leaveBreadcrumb(breadcrumb: string): void;
   notifyError(error: Error): void;
+  attachErrorListeners(): void;
 }
+
+export type ErrorState = {
+  severity: string;
+  unhandled: boolean;
+  severityReason: { type: string };
+};
+
+export type EventTarget = {
```

Comment:
we can extend the EventTarget interface like

export interface ErrorTarget extends EventTarget or use the EventTarget directly if it cover us

Reply:
Extended Element